### PR TITLE
DAOS-15596 pkg: Update argobots to 1.2

### PR DESCRIPTION
**Pull request by knard-intel**
### Description

Update argopbots dependency to 1.2 release

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by knard-intel** [comment]
**WARNING** This PR should not be merged before the PR https://github.com/daos-stack/argobots/pull/26 of the argobots-package one.

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -592,6 +592,9 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Tue Sep 24 2024 Cedric Koch-Hofer <cedric.koch-hofer@intel.com> 2.7.100-6
+- Update argobots to 1.2
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:596]:
```suggestion
- Update BR: argobots to 1.2
```

File: utils/rpms/daos.spec [hunk 2]
```diff
@@ -41,7 +41,7 @@ BuildRequires: hwloc-devel
 BuildRequires: bullseye
 %endif
 %if (0%{?rhel} >= 8)
-BuildRequires: argobots-devel >= 1.1
+BuildRequires: argobots-devel >= 1.2
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:44]: You also need to update https://github.com/daos-stack/daos/blob/861445b7ff2d49880a4df4f71f4e5ddd6279208e/utils/rpms/daos.spec#L48

But probably a better way to handle this is to add a new 
```
%global argobots_version 1.2
```

after https://github.com/daos-stack/daos/blob/861445b7ff2d49880a4df4f71f4e5ddd6279208e/utils/rpms/daos.spec#L7 and then you can do this here:
```suggestion
BuildRequires: argobots-devel >= %{argobots_version}
```

and then

```patch
-BuildRequires: libabt-devel >= 1.0rc1
+BuildRequires: libabt-devel >= %{argobots_version}
```

on line 48.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -592,6 +592,9 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Tue Sep 24 2024 Cedric Koch-Hofer <cedric.koch-hofer@intel.com> 2.7.100-6
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec:595]: Please freshen the datestamp here.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
You should also update https://github.com/daos-stack/daos/blob/861445b7ff2d49880a4df4f71f4e5ddd6279208e/debian/control#L19 similarly and add the necessary `debian/changelog` entry update.

File: utils/rpms/daos.spec (hunk 2 above)
Comment (knard-intel) [reply utils/rpms/daos.spec:44]: - [ ] Improve RPM build dep version handling

File: utils/rpms/daos.spec (hunk 3 above)
Comment (knard-intel) [reply utils/rpms/daos.spec:595]: - [ ] Refresh changelong timestamp

File: utils/rpms/daos.spec
Comment (brianjmurrell) [inline utils/rpms/daos.spec:1]: You also need to update line 49 with `%{argobots_version}` as such:
```patch
-BuildRequires: libabt-devel >= 1.0rc1
+BuildRequires: libabt-devel >= %{argobots_version}
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: utils/rpms/daos.spec
Comment (knard-intel) [reply utils/rpms/daos.spec:1]: Oups missed this one... Thanks for the awk eyes :-)

- [ ] Fix missing version update

Comment (knard-intel) [reply utils/rpms/daos.spec:1]: > Oups missed this one... Thanks for the awk eyes :-)
> 
> * [x]  Fix missing version update

Fixed with commit 446561a0c

**Review by brianjmurrell — DISMISSED** [review]

**Comment by knard-intel** [comment]
@brianjmurrell could you tell me, please, if the PR is OK for you of if something is missing.

**Review by brianjmurrell — APPROVED** [review]

**Comment by knard-intel** [comment]
@daos-stack/daos-gatekeeper , please could you lend this PR with the following commit message:
Title: `DAOS-15596 pkg: Update argobots to 1.2`
Body:
`Update argopbots dependency to 1.2 release`

**Pull request merged**